spyder_unittest/backend/testrunner.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Spyder Project Developers
+#
+# Licensed under the terms of the MIT License
+# (see LICENSE.txt for details)
+# -----------------------------------------------------------------------------
+"""Classes for running tests within various frameworks."""
+
+# Standard library imports
+from collections import namedtuple
+import os
+import tempfile
+
+# Third party imports
+from lxml import etree
+from qtpy.QtCore import (QObject, QProcess, QProcessEnvironment, QTextCodec,
+                         Signal)
+from spyder.py3compat import to_text_string
+from spyder.utils.misc import add_pathlist_to_PYTHONPATH
+
+# Class representing test results
+TestResult = namedtuple('TestResult', [
+    'category', 'status', 'name', 'message', 'time', 'extra_text'
+])
+
+
+class Category:
+    """Enum type representing category of test result."""
+
+    OK = 1
+    FAIL = 2
+    SKIP = 3
+
+
+STATUS_TO_CATEGORY = {
+    'ok': Category.OK,
+    'failure': Category.FAIL,  # py.test
+    'error': Category.FAIL,  # nose
+    'skipped': Category.SKIP,  # py.test, nose
+}
+
+
+class TestRunner(QObject):
+    """
+    Class for running tests with py.test or nose.
+
+    All communication back to the caller is done via signals.
+
+    Fields
+    ------
```

Comment:
I think the usual name is:

```python
Parameters
-----------
```

At least for numpy doc (which looks like what you are using :-p)